docs/source/admin_guide/prefect.md

```diff
@@ -33,6 +33,48 @@ The `TF_VAR_prefect_token` API key is set as `PREFECT__CLOUD__AGENT__AUTH_TOKEN`
 environment variable in the agent. It is used while deploying Prefect Agent so that
 it can connect to Prefect Cloud and query flows.
 
+## Configuration overrides
+You can override your agent configurations without having to modify the cluster configuration files.  The extra variable `overrides` makes this
+possible by changing the default values for the Agent based on an overriding setting presented on your qhub-config.yaml file.
+
+The current variables that can be overridden include:
+
+```
+- IMAGE_PULL_SECRETS
+- PREFECT__CLOUD__AGENT__LABELS
+- PREFECT__CONTEXT__SECRETS__GCP_CREDENTIALS
+- JOB_MEM_REQUEST
+- JOB_MEM_LIMIT
+- JOB_CPU_REQUEST
+- JOB_CPU_LIMIT
+- IMAGE_PULL_POLICY
+```
+
+For example, if you just want to override the amount of CPU limits for each job, you would need to craft a declarative configuration,
+in you qhub-config.yaml file, as follows:
+
+```yaml
+prefect:
+  enabled: true
+  overrides:
+      prefect_agent:
+        job:
+          resources:
+            limit:
+              cpu: 4
+```
+Also, if you would like to include an extra variable to the agent environment configuration, you can do it by including it under
+the `extraEnvVars` field in the overrides block. For example, when using local secrets, your Prefect Agent can be configured
+to authenticate to GCP automatically by adding that specific `GCP_CREDENTIALS` key value pair into your secrets context like so:
+
+```yalm
+prefect:
+  enabled: true
+  overrides:
+    extraEnvVars:
+        PREFECT__CONTEXT__SECRETS__GCP_CREDENTIALS: '<Your value>'
+```
+It's important to keep in mind that all variables defined under `extraEnvVars` must be base64 encoded before passing it's value to the qhub-config file.
 
```

Comment:
Why do they all need to be base64 encoded?  Is there no way around that?  Seems very inconvenient as a developer.

Reply:
I agree with you in that, unfortunately every time we pass the correct structure via cookiecutter to terraform, it understands it as a string, but if the variable has some special characteristics, it is a list or a json, the helm chart will not parse the string correctly resulting in some errors as
```
Error: ---> error unmarshaling JSON: while decoding JSON: json: cannot unmarshal array into Go value of type map[string]interface {} [{'PREFECT__CLOUD__AGENT__LABELS': ['dev', 'prod'], 'JOB_MEM_REQUEST': '20Gi'}]
```

Reply:
> a rush to get this PR in (are we?)

Nope. We are rebasing onto this branch so no pressure to merge the PR.

Reply:
I made some progress on this.  I'll push the updates in a commit tomorrow.